src/server/publicApi.js

```diff
@@ -1,125 +1,181 @@
+                        if (c.id === componentIdx) {
+                            cell = c;
+                            break;
+                        }
+                    }
+                    if (cell !== null) {
+                        let threats = [];
+                        if (Array.isArray(cell.threats)) {
+                            threats = cell.threats;
+                        }
+                        Object.keys(
+                            res.G.identifiedThreats[diagramIdx][componentIdx]
+                        ).forEach((threatIdx) => {
+                            let t =
+                                res.G.identifiedThreats[diagramIdx][componentIdx][threatIdx];
+                            threats.push({
+                                status: "Open",
+                                severity: t.severity,
+                                id: t.id,
+                                methodology: "STRIDE",
+                                type: getTypeString(t.type),
+                                title: t.title,
+                                description: t.description,
+                                mitigation: t.mitigation,
+                                owner: metadata.players[t.owner].name,
+                                game: gameID,
+                            });
+                        });
+                        cell.threats = threats;
+                    }
+                }
+            );
         });
+
+        const modelTitle = model.summary.title.replace(" ", "-");
+        const timestamp = new Date().toISOString().replace(":", "-");
+        ctx.attachment(`${modelTitle}-${timestamp}.json`);
+        ctx.body = model;
     });
 
-    const modelTitle = model.summary.title.replace(' ', '-');
-    const timestamp = (new Date()).toISOString().replace(":", "-");   
-    ctx.attachment(`${modelTitle}-${timestamp}.json`);
-    ctx.body = model;
+
+    //produce a nice textfile with the threats in
+    router.get("/download/text/:id", async (ctx) => {
+        //get some variables that might be useful
+        const gameName = ElevationOfPrivilege.name;
+        const gameID = ctx.params.id;
+        const res = await gameServer.db.get(`${gameName}:${gameID}`);
+        var threats = [];
+
+        Object.keys(res.G.identifiedThreats).forEach((diagramId) => {
+            Object.keys(res.G.identifiedThreats[diagramId]).forEach((componentId) => {
+                Object.keys(res.G.identifiedThreats[diagramId][componentId]).forEach((threatId) => {
+                    threats.push(res.G.identifiedThreats[diagramId][componentId][threatId]);
+                });
+            });
+        });
+        
+        const timestamp = new Date().toISOString().replace(":", "-");
+        ctx.attachment(`threats-${timestamp}.txt`);
+        //ctx.body = threats;
+
+        //copied formatting from christoph's script
+        
+        ctx.body = `Threats
```

Comment:
Now that I think of it, maybe it's a good idea to create this textfile in markdown syntax. Then its readable by most tools. What do you think?

Reply:
Yeah, I'll do that. It'll make it a bit nicer to look at.

I'm just thinking of how I'll format it. Let me know if anything will make this more readable. This is what I have at the moment:
another suggestion might get rid of the lists and just have paragraphs, or to use a table or something

Threats
=======

**1. Threat One**

  - *Author:* 		Player 1

  - *Description:*	An attacker can take advantage of your custom key exchange or integrity control which you built instead of using standard crypto


**2. Threat Two**

  - *Author:* 		Player 2

  - *Description:*	An attacker can take advantage of your custom key exchange or integrity control which you built instead of using standard crypto

  - *Mitigation:*	sadfasdfa

Reply:
I like it that way! :-D

Reply:
Maybe in the title you could add the date.